api/v1alpha3/awsmachine_webhook.go

```diff
@@ -38,12 +39,21 @@ func (r *AWSMachine) SetupWebhookWithManager(mgr ctrl.Manager) error {
 }
 
 // +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1alpha3-awsmachine,mutating=false,failurePolicy=fail,groups=infrastructure.cluster.x-k8s.io,resources=awsmachines,versions=v1alpha3,name=validation.awsmachine.infrastructure.cluster.x-k8s.io
+// +kubebuilder:webhook:verbs=create;update,path=/mutate-infrastructure-cluster-x-k8s-io-v1alpha3-awsmachine,mutating=true,failurePolicy=fail,groups=infrastructure.cluster.x-k8s.io,resources=awsmachines,versions=v1alpha3,name=default.awsmachine.infrastructure.cluster.x-k8s.io
 
 var _ webhook.Validator = &AWSMachine{}
+var _ webhook.Defaulter = &AWSMachine{}
+
+// Default implements webhook.Defaulter so a mutating webhook will be registered for the type
+func (r *AWSMachine) Default() {
+	if r.Spec.ProviderID == nil && r.Spec.CloudInit.Encrypted == nil {
```

Comment:
If encrypted is nil, set it to true. Shouldn't need to check provider id.

Reply:
Actually, shouldn't CloudInit be a pointer on the v1alpha2 type, and not the v1alpha3 one?